src/Plugin/NoWaitForResponseRequest.php

```diff
@@ -0,0 +1,139 @@
+
+use Solarium\Core\Client\Adapter\ConnectionTimeoutAwareInterface;
+use Solarium\Core\Client\Adapter\Curl;
+use Solarium\Core\Client\Adapter\TimeoutAwareInterface;
+use Solarium\Core\Client\Request;
+use Solarium\Core\Client\Response;
+use Solarium\Core\Event\Events;
+use Solarium\Core\Event\PreExecuteRequest;
+use Solarium\Core\Plugin\AbstractPlugin;
+use Solarium\Exception\HttpException;
+
+/**
+ * NoWaitForResponseRequest plugin.
+ *
+ * Long-running requests like suggest.buildAll might exceed timeouts.
+ * This plugin "tries" to convert the request in a kind of fire-and-forget.
+ * Most reliable if using the Curl adapter.
+ */
+class NoWaitForResponseRequest extends AbstractPlugin
+{
+    /**
+     * Event hook to adjust client settings just before query execution.
+     *
+     * @param object $event
+     *
+     * @return self Provides fluent interface
+     *
+     * @throws \Solarium\Exception\HttpException
+     */
+    public function preExecuteRequest($event): self
+    {
+        // We need to accept event proxies or decorators.
+        /** @var PreExecuteRequest $event */
+        $request = $event->getRequest();
+        $queryString = $request->getQueryString();
+
+        if (Request::METHOD_GET === $request->getMethod()) {
+            // GET requests usually expect a result. Since the purpose of this
+            // plugin is to trigger a long-running command and to not wait for
+            // its result, POST is the correct method.
+            // Depending on the HTTP configuration, GET requests could be
+            // cached. If this plugin is used, someone usually wants to build a
+            // dictionary or suggester and caching has to be avoided. Even if
+            // Solr accepts GET requests for these tasks, POST is the correct
+            // method.
+            $charset = $request->getParam('ie') ?? 'utf-8';
+            $request->setMethod(Request::METHOD_POST);
+            $request->setContentType(Request::CONTENT_TYPE_APPLICATION_X_WWW_FORM_URLENCODED, ['charset' => $charset]);
+            $request->setRawData($queryString);
+            $request->clearParams();
+        }
+
+        $timeout = TimeoutAwareInterface::DEFAULT_TIMEOUT;
+        if ($this->client->getAdapter() instanceof TimeoutAwareInterface) {
+            $timeout = $this->client->getAdapter()->getTimeout();
+            if (($this->client->getAdapter() instanceof ConnectionTimeoutAwareInterface) && ($this->client->getAdapter()->getConnectionTimeout() > 0)) {
+                $this->client->getAdapter()->setTimeout($this->client->getAdapter()->getConnectionTimeout() + TimeoutAwareInterface::FAST_TIMEOUT);
+            } else {
+                $this->client->getAdapter()->setTimeout(TimeoutAwareInterface::FAST_TIMEOUT);
+            }
+        }
+
+        if ($this->client->getAdapter() instanceof Curl) {
+            $this->client->getAdapter()->setOption('return_transfer', false);
+        }
+
+        $microtime1 = microtime();
```

Comment:
I think this should be `microtime(true)`.

E.g. https://onlinephp.io?s=s7EvyCjg5eLlKkssik8pzS3QUCnJzE01VLBVyM1MLsoHcTQ0Na2RFaRhqigpKk0Fq-LlKs5JTS3QMIFw4JqQDFPQVYDYgWIqmlkgRWlQVQA%2C&v=8.2.13

```php
<?php

var_dump($time1 = microtime());
var_dump($ftime1 = microtime(true));

sleep(4);

var_dump(microtime() - $time1);
var_dump(microtime(true) - $ftime1);
```

```
string(21) "0.33706600 1704466431"
float(1704466431.337109)

Warning: A non-numeric value encountered in /home/user/scripts/code.php on line 8

Warning: A non-numeric value encountered in /home/user/scripts/code.php on line 8
float(0.00018300000000004424)
float(4.000219821929932)
```